tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php

```diff
@@ -5904,15 +5904,15 @@ public function dataFunctions(): array
 				'$strSplitConstantStringWithFailureSplitLength',
 			],
 			[
-				PHP_VERSION_ID < 80000 ? 'array<int, string>|false' : 'array<int, string>',
+				PHP_VERSION_ID < 80000 ? '(array<int, string>&nonEmpty)|false' : 'array<int, string>&nonEmpty',
 				'$strSplitConstantStringWithInvalidSplitLengthType',
 			],
 			[
 				'array<int, string>&nonEmpty',
 				'$strSplitConstantStringWithVariableStringAndConstantSplitLength',
 			],
 			[
-				PHP_VERSION_ID < 80000 ? 'array<int, string>|false' : 'array<int, string>',
+				PHP_VERSION_ID < 80000 ? '(array<int, string>&nonEmpty)|false' : 'array<int, string>&nonEmpty',
```

Comment:
interessting.. sometimes a typo including braces is produced, and sometimes without

Reply:
https://github.com/phpstan/phpstan-src/blob/119facc64db3acd77b8024fafa8bb3228e90cb25/src/Type/UnionType.php#L150-L168